bundles/org.eclipse.ui.ide/src/org/eclipse/ui/views/markers/internal/ContentGeneratorDescriptor.java

```diff
@@ -227,14 +267,12 @@ public MarkerType getType(String typeId) {
 	 * @return Map of {@link String} to {@link MarkerType}
 	 */
 	public Map<String, MarkerType> getTypesTable() {
-		if (allTypesTable == null) {
```

Comment:
This again removes a cache. Is that necessary?

Reply:
Not, and shouldn't be removed.